### PR TITLE
#25 Faculty details

### PR DESCRIPTION
**Pull request by mikolaj-jalocha**
[Screen_recording_20240720_210728.webm](https://github.com/user-attachments/assets/e71fbff7-1777-4e49-8fa0-9648dd45f7cd)

### CODE REVIEW

File: lib/features/departments_tab/widgets/department_card.dart [hunk 1]
```diff
@@ -9,11 +9,13 @@ import '../../../widgets/wide_tile_card.dart';
 
 class DepartmentCard extends StatelessWidget {
   final Department department;
-  const DepartmentCard(this.department, {super.key});
+  const DepartmentCard(this.department, {super.key, this.onClick});
+  final void Function()? onClick;
```
Comment (simon-the-shark) [inline lib/features/departments_tab/widgets/department_card.dart:13]: maybe `VoidCallback` - not forcing though?

File: lib/features/home_view/widgets/departments_section/deparment_box.dart [hunk 2]
```diff
@@ -10,7 +10,8 @@ import '../../../../widgets/tile_splash.dart';
 
 class DepartmentBox extends StatelessWidget {
   final Department department;
-  const DepartmentBox(this.department, {super.key});
+  final void Function() onClick;
```
Comment (simon-the-shark) [inline lib/features/home_view/widgets/departments_section/deparment_box.dart:13]: same maybe?

File: lib/widgets/details_screen_contact_section.dart [hunk 3]
```diff
@@ -61,7 +61,7 @@ class _IconWithUrl extends StatelessWidget {
               text: TextSpan(
                 text: text,
                 style: context.textTheme.bodyOrange
-                    .copyWith(decoration: TextDecoration.underline),
+                    .copyWith(decoration: url.isNotEmpty ? TextDecoration.underline : TextDecoration.none),
```
Comment (simon-the-shark) [inline lib/widgets/details_screen_contact_section.dart:64]: nice

**Review by simon-the-shark — COMMENTED** [review]
imo, pretty neat and cool code. Few really small notes, without any greater reservations. Good job! 🦈

File: lib/features/department_details/department_details.dart [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                    List.empty()),
+            FieldsOfStudySection(
+              fieldsOfStudy: value?.Departments_by_id?.fieldsOfStudies
+                      ?.map((e) => e?.name ?? "")
+                      .toList() ??
+                  List.empty(),
+            ),
+            DepartmentsStudyCirclesSection(
+                value?.Scientific_Circles.whereNotNull().toList() ??
+                    List.empty()),
```
[the remark this answers is not in the document]
Comment (mikolaj-jalocha) [reply lib/features/department_details/department_details.dart:82]: Wipe your tears! WhereNonNull is now used :-)

File: lib/features/department_details/widgets/department_study_circle_section.dart [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import '../../../widgets/big_preview_card.dart';
+import '../../../widgets/subsection_header.dart';
+import '../../home_view/widgets/paddings.dart';
+import '../../navigator/navigator/detail_view_navigator.dart';
+import '../../navigator/navigator/nested_navigator.dart';
+import '../../navigator/navigator/tab_bar_navigator.dart';
+import '../repositories/department_details_repository.dart';
+
+
+// TODO: Resolve if the list button should redirect to list of all study circles or only ones related to the department?
```
[the remark this answers is not in the document]
Comment (mikolaj-jalocha) [reply lib/features/department_details/widgets/department_study_circle_section.dart:16]: I will mention it on our weekly

File: lib/features/departments_tab/widgets/department_card.dart (hunk 1 above)
Comment (mikolaj-jalocha) [reply lib/features/departments_tab/widgets/department_card.dart:13]: I didn't know about it! Nice and clean actually

**Review by simon-the-shark — APPROVED** [review]

**Pull request merged**